src/Catalyst.Core.Modules.Dfs.Tests/CoreApi/BitswapApiTest.cs

```diff
@@ -0,0 +1,433 @@
+                    await Task.Delay(100);
+                    var w = await ipfs.Bitswap.WantsAsync();
+                    if (w.Contains(block.Id))
+                        break;
+                }
+
+                cts.Cancel();
+                var wants = await ipfs.Bitswap.WantsAsync();
+                wants.ToArray().Should().NotContain(block.Id);
+                Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task Unwant()
+        {
+            await ipfs.StartAsync();
+            try
+            {
+                var block = new DagNode(Encoding.UTF8.GetBytes("BitswapApiTest unknown block 2"));
+                Task wantTask = ipfs.Bitswap.GetAsync(block.Id);
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                        throw new Xunit.Sdk.XunitException("wanted block is missing");
+
+                    // Assert.Fail("wanted block is missing");
+                    await Task.Delay(100);
+                    var w = await ipfs.Bitswap.WantsAsync();
+                    if (w.Contains(block.Id))
+                        break;
+                }
+
+                await ipfs.Bitswap.UnwantAsync(block.Id);
+                var wants = await ipfs.Bitswap.WantsAsync();
+                wants.ToArray().Should().NotContain(block.Id);
+                Assert.True(wantTask.IsCanceled);
+            }
+            finally
+            {
+                await ipfs.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task OnConnect_Sends_WantList()
+        {
+            ipfs.Options.Discovery.DisableMdns = true;
+            ipfs.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await ipfs.StartAsync();
+
+            ipfsOther.Options.Discovery.DisableMdns = true;
+            ipfsOther.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+            await ipfsOther.StartAsync();
+            try
+            {
+                var local = await ipfs.LocalPeer;
+                var remote = await ipfsOther.LocalPeer;
+                Console.WriteLine($"this at {local.Addresses.First()}");
+                Console.WriteLine($"othr at {remote.Addresses.First()}");
+
+                var data = Guid.NewGuid().ToByteArray();
+                var cid = new Cid {Hash = MultiHash.ComputeHash(data)};
+                var _ = ipfs.Block.GetAsync(cid);
+                await ipfs.Swarm.ConnectAsync(remote.Addresses.First());
+
+                var endTime = DateTime.Now.AddSeconds(10);
+                while (DateTime.Now < endTime)
+                {
+                    var wants = await ipfsOther.Bitswap.WantsAsync(local.Id);
+                    if (wants.Contains(cid))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)